file_process/file_processor_factory.py

```diff
@@ -6,10 +6,23 @@
 
 
 class FileProcessFactory:  # pylint: disable=too-few-public-methods
+    EXTENSIONS_MAP = {
+        '.h5ad': H5ADFileProcessor,
+        '.csv': CSVFileProcessor
+    }
+
     @classmethod
     def get(cls, filename: str, file: BytesIO, **kwargs):
-        if filename.endswith('.h5ad'):
-            return H5ADFileProcessor(file, **kwargs)
-        if filename.endswith('.csv'):
-            return CSVFileProcessor(file, **kwargs)
+        for extension, processor_class in cls.EXTENSIONS_MAP.items():
+            if filename.endswith(extension):
```

Comment:
Should the file extension be not case sensitive? I think it will be good to cast the file name to lowercase before the comparison, so that the code covers the filenames in upper or mixed cases.

Reply:
Idk, filenames and extensions are case-sensitive in linux and windows, but non-case-sensitive in macos. I feel it as more natural for extensions to be case sensitive

Reply:
ok